### PR TITLE
Cvs 35968 better handling of anonymous shape param

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/modelconfig.cpp [hunk 1]
```diff
@@ -358,7 +358,11 @@ Status ModelConfig::parseNode(const rapidjson::Value& v) {
                             shapeInfo.shape.push_back(sh.GetUint64());
                         }
                     }
-                    this->addShape(s.name.GetString(), shapeInfo);
+                    if (s.name.GetString() != ANONYMOUS_INPUT_NAME) {
+                        this->addShape(s.name.GetString(), shapeInfo);
+                    } else {
+                        SPDLOG_WARN("Provided one of provided shapes name: {} is forbidden and will be omitted", ANONYMOUS_INPUT_NAME);
```
Comment (rasapala) [inline src/modelconfig.cpp:364]: Provided shape name: {} is forbidden

**Review by rasapala — CHANGES_REQUESTED** [review]

**Review by rasapala — APPROVED** [review]
LGTM

File: src/test/modelconfig_test.cpp [hunk 2]
```diff
@@ -439,3 +439,32 @@ TEST(ModelConfig, modelVersionPolicyIncorrect) {
...
 }
+
+TEST(ModelConfig, ConfigParseNodeWithForbiddenShapeName) {
+    const char* config = R"#({
+        "model_config_list": [
+            {
+                "config": {
+                    "name": "alpha",
+                    "base_path": "/tmp/models/dummy1",
+                    "shape": {"ANONYMOUS_INPUT_NAME": "(1, 3, 600, 600)"}
```
Comment (dkalinowski) [inline src/test/modelconfig_test.cpp:450]: Can we somehow avoid hardcoding ANONYMOUS_INPUT_NAME? We already have it defined somewhere else under alias `ANONYMOUS_INPUT_NAME`. Maybe we could somehow use it here?

**Review by dkalinowski — CHANGES_REQUESTED** [review]
Please also create task for long term solution - without forbidding any input name.

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**